src/containers/quiz/quiz-question/Question.constants.ts

```diff
@@ -0,0 +1,45 @@
+import { questionType } from '~/components/question-editor/QuestionEditor.constants'
+import { Question } from '~/types'
+
+interface GetQuestionStatusParams {
+  question: Question
+  answer: null | string | string[]
+  showAnswersCorrectness: boolean
+}
+
+export const isCorrectAnswer = (
+  question: Question,
+  answer: null | string | string[]
+) => {
+  if (!answer) return false
+  const { isMultipleChoice, isOpenAnswer } = questionType(question.type)
+
+  if (isOpenAnswer) {
+    return true
+  } else if (isMultipleChoice && Array.isArray(answer)) {
+    const correctAnswers = question.answers.filter((item) => item.isCorrect)
+
+    return (
+      answer.every(
+        (item) => question.answers.find((i) => i.text === item)?.isCorrect
+      ) && answer.length === correctAnswers.length
```

Comment:
I think we should add check for equal length before looping using `answer.every`. I hope it will improve performance on large list of questions

Reply:
Also letters letters don't cost any money:) So i suggest using `answer` variable name instead of `i`

Reply:
I moved check for length before loop and made variable more descriptive here and in other place